### PR TITLE
Update patches to external config 

### PR DESCRIPTION
**Pull request by TomasVotruba**
Ref https://symfony.com/blog/symfony-6-0-0-released

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -38,13 +38,13 @@
...
         "symfony/contracts": "^2.5",
-        "symfony/dependency-injection": "^5.3|^6.0",
+        "symfony/dependency-injection": "6.0 as 5.4",
         "symfony/deprecation-contracts": "^2.4",
-        "symfony/finder": "^5.3|^6.0",
-        "symfony/process": "^5.3|^6.0",
-        "symfony/yaml": "^5.3|^6.0",
+        "symfony/finder": "6.0 as 5.4",
+        "symfony/process": "6.0 as 5.4",
+        "symfony/yaml": "6.0 as 5.4",
```
Comment (TomasVotruba) [inline composer.json]: The last current blocker is php-cs-fixer https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6095

**Comment by samsonasik** [comment]
@TomasVotruba it seems it cause error in main branch on packages-tests even after restart CI build:

```bash


There were 3 errors:

1) Rector\PHPUnit\Tests\Rector\Class_\ArrayArgumentToDataProviderRector\ArrayArgumentToDataProviderRectorTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
BadMethodCallException: Call to undefined method "Symfony\Component\DependencyInjection\Loader\Configurator\ServiceConfigurator::configure()".
```

https://github.com/rectorphp/rector-phpunit/runs/4443637106?check_suite_focus=true#step:5:16

**Comment by samsonasik** [comment]
@TomasVotruba it seems due to enable-patching config in rector-* packages ? eg: https://github.com/rectorphp/rector-phpunit/commit/1914ca5117f1c8d6e22171b4a1eb16e05994693a

**Comment by TomasVotruba** [comment]
I'm working on this one...

<br>

So far I've found:

* patch-file does not work for dependencies

:red_circle: 

* this is needed in `rector/rector-src` for dependencies

```json
"extra": {
        "patches": {
            "package-name": ["patches/some.patch"]
        }  
}
```

:red_circle: 

* but relative path are resolved incorrectly, as `rector/rector-src` is nested - it seems like a bug in the composer patching packages, as there is no other way to use it 

```json
"extra": {
        "patches": {
            "package-name": ["https//raw.github.com/.../patches/some.patch"]
        }  
}
```

:heavy_check_mark:

**Comment by samsonasik** [comment]
@TomasVotruba  thank you 👍

**Comment by TomasVotruba** [comment]
We're green again on packages: https://github.com/rectorphp/rector-src/blob/main/dev-docs/packages_ci_status.md

**Comment by TomasVotruba** [comment]
I've also added little docs in case this will break in the future :)

https://github.com/rectorphp/vendor-patches/#how-to-use

**Pull request merged**